src/framework/parsers/draco-worker.js

```diff
@@ -0,0 +1,259 @@
+function DracoWorker(jsUrl, wasmUrl) {
+    // load the wasm module and return a promise which will resolve
+    // to the emscripten module instance
+    const loadWasm = (moduleName, jsUrl, wasmUrl) => {
+        // import js
+        importScripts(jsUrl);
+
+        // instantiate the module
+        return self[moduleName]({ locateFile: () => wasmUrl })
+            .then(instance => instance);
+    };
+
+    const wrap = (typedArray, dataType) => {
+        switch (dataType) {
+            case draco.DT_UINT8: return new Uint8Array(typedArray.buffer, typedArray.byteOffset, typedArray.byteLength);
+            case draco.DT_UINT16: return new Uint16Array(typedArray.buffer, typedArray.byteOffset, typedArray.byteLength / 2);
+            case draco.DT_UINT32: return new Uint32Array(typedArray.buffer, typedArray.byteOffset, typedArray.byteLength / 4);
+            case draco.DT_FLOAT32: return new Float32Array(typedArray.buffer, typedArray.byteOffset, typedArray.byteLength / 4);
+        }
+        return null;
+    }
+
+    const componentSizeInBytes = (dataType) => {
+        switch (dataType) {
+            case draco.DT_UINT8: return 1;
+            case draco.DT_UINT16: return 2;
+            case draco.DT_UINT32: return 4;
+            case draco.DT_FLOAT32: return 4;
+        }
+        return 1;
+    };
+
+    const attributeSizeInBytes = (attribute) => {
+        return attribute.num_components() * componentSizeInBytes(attribute.data_type());
+    };
+
+    const attributeOrder = {
```

Comment:
maybe we should send this table to the worker as part of the job description, to avoid hardcoded constants?

Reply:
these are draco constants, not engine constants unfortunately.